### PR TITLE
PR to add some hash problems

### PR DESCRIPTION
**Pull request by danghai**
(Put an `X` inside the `[ ]` to denote check mark `[X]`.)

- [x] **If creating a new file :**
  - [x] added links to it in the README files ?
  - [x] included tests with it ?
  - [x] added description (overview of algorithm, time and space compleixty, and possible edge case) in docstrings ?

- [ ] **if done some changes :**
  - [ ] wrote short description in the PR explaining what the changes do ?
  - [ ] Fixes #[issue number] if related to any issue

- [ ] **other**

### CODE REVIEW

File: algorithms/map/is_isomorphic.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    :type s: str
+    :type t: str
+    :rtype: bool
+    """
+    if len(s) != len(t):
+        return False
+    map = {}
+    for i in range(len(s)):
+        if s[i] not in map:
+            if t[i] in map.values():
```
Comment (goswami-rahul) [inline algorithms/map/is_isomorphic.py]: We can make this O(n) from O(n^2), if we store the map.values() in a seperate set.

File: algorithms/map/is_isomorphic.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+"""
+def is_isomorphic(s, t):
+    """
+    :type s: str
+    :type t: str
+    :rtype: bool
+    """
+    if len(s) != len(t):
+        return False
+    map = {}
```
Comment (goswami-rahul) [inline algorithms/map/is_isomorphic.py]: map overshadows built-in `map` in Python.

File: algorithms/map/word_pattern.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+Reference: https://leetcode.com/problems/word-pattern/description/
+"""
+def word_pattern(pattern, str):
+    map = {}
+    list_str = str.split()
+    if len(list_str) != len(pattern):
+        return False
+    for i in range(len(pattern)):
+        if pattern[i] not in map:
+            if list_str[i] in map.values():
```
Comment (goswami-rahul) [inline algorithms/map/word_pattern.py]: Similar optimization can be used here too!

File: algorithms/set/find_keyboard_row.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+Reference: https://leetcode.com/problems/keyboard-row/description/
+"""
+def find_keyboard_row(words):
+    """
+    :type words: List[str]
+    :rtype: List[str]
+    """
+    keyboard = [
+        set(list('qwertyuiop')),
```
Comment (goswami-rahul) [inline algorithms/set/find_keyboard_row.py]: I think `set('qwertyuiop')` will be fine. No need for `list`.

Comment (keon) [reply algorithms/set/find_keyboard_row.py]:
```
>>> set('acvvvv')
set(['a', 'c', 'v'])
```
yep

Comment (danghai) [reply algorithms/set/find_keyboard_row.py]: Okay, I will fix now

File: algorithms/map/is_isomorphic.py (hunk 2 above)
Comment (keon) [reply algorithms/map/is_isomorphic.py]: yep good catch

File: algorithms/map/word_pattern.py (hunk 3 above)
Comment (danghai) [reply algorithms/map/word_pattern.py]: It is a good idea

**Comment by danghai** [comment]
@goswami-rahul @keon I have updated!

**Comment by keon** [comment]
nice work :)
